### PR TITLE
Reduce reliance on `data_*` throughout LBANN (1/?)

### PR DESCRIPTION
**Pull request by tbennun**

### CODE REVIEW

File: src/callbacks/summarize_images.cpp [hunk 1]
```diff
@@ -230,69 +230,8 @@ build_categorical_accuracy_strategy_from_pbuf(
...
-      if (m_tracked_images.find(sample_index) != m_tracked_images.end()) {
-        img_indices.push_back(std::make_pair(ii, sample_index));
-      }
-      else if (m_tracked_images.size() < m_num_images) {
-        m_tracked_images.insert(sample_index);
-        img_indices.push_back(std::make_pair(ii, sample_index));
-      }
-    }
-  }
+  LBANN_ERROR("This feature is no longer supported");
```
Comment (bvanessen) [inline src/callbacks/summarize_images.cpp:234]: @tbennun Is this just to clean it up or is there something that is broken in this one or by this PR.

Comment (tbennun) [reply src/callbacks/summarize_images.cpp:234]: It is just cleanup, nothing broken here.

Comment (benson31) [reply src/callbacks/summarize_images.cpp:234]: So I recognize that this is extremely hacky code, but do we just want to remove this capability?

**Review by benson31 — APPROVED** [review]

File: src/callbacks/summarize_images.cpp (hunk 1 above)
Comment (tbennun) [reply src/callbacks/summarize_images.cpp:234]: we can bring it back to the codebase and get rid of it in a callback cleanup PR `(4/?)`?

**Pull request merged**